docs/src/examples.md

```diff
@@ -86,6 +86,45 @@ julia> round.(vcov(ols); digits=5)
   0.38889  -0.16667
  -0.16667   0.08333
 ```
+By default, the `lm` method uses the `Cholesky` factorization which is known as fast but numerically unstable, especially for ill-conditioned design matrices. You can use the `method` keyword argument to apply a more stable `QR` factorization method.
+
+```jldoctest
+julia> data = DataFrame(X=[1,2,3], Y=[2,4,7]);
+
+julia> ols = lm(@formula(Y ~ X), data; method=:qr)
+LinearModel
+
+Y ~ 1 + X
+
+Coefficients:
+─────────────────────────────────────────────────────────────────────────
+                 Coef.  Std. Error      t  Pr(>|t|)  Lower 95%  Upper 95%
+─────────────────────────────────────────────────────────────────────────
+(Intercept)  -0.666667    0.62361   -1.07    0.4788   -8.59038    7.25704
+X             2.5         0.288675   8.66    0.0732   -1.16797    6.16797
+─────────────────────────────────────────────────────────────────────────
+```
+Suppose we have `y = [1, 2, 3, 4, 5]` and `x = [1.0E-12, 2.0E-12, 3.0E-12, 4.0E-12, 5.0E-12]`.
+Clearly y = 0 + 1.0E12 * x. So if we fit a linear model `y ~ x` then the estimate of the intercept should be `0` and the estimate of slop should be `1.0E12`.
+The following example shows that `QR` decomposition works better for ill-conditioned design matrix. The linear model with the `Cholesky` decomposition method is unable to estimate parameters correctly whereas the linear model with the `QR` decomposition does.
+
+```jldoctest
+julia> y = [1, 2, 3, 4, 5];
+
+julia> x = [1.0E-12, 2.0E-12, 3.0E-12, 4.0E-12, 5.0E-12];
+
+julia> nasty = DataFrame(y = y, x = x);
+
+julia> mdl1 = lm(@formula(y ~ x), nasty; method=:cholesky);
```

Comment:
then we should show here the default behavior (which should produce a correct result now)

Reply:
The problem is that the estimates (small numbers) differ in different systems.  Initially, we added the following: -
```
julia> y = [1, 2, 3, 4, 5];
julia> x = [1.0E-12, 2.0E-12, 3.0E-12, 4.0E-12, 5.0E-12];
julia> nasty = DataFrame(y = y, x = x);
julia> lm(@formula(y ~ x), nasty; method=:cholesky)
LinearModel
y ~ 1 + x
Coefficients:
──────────────────────────────────────────────────────────────────────
             Coef.  Std. Error       t  Pr(>|t|)  Lower 95%  Upper 95%
──────────────────────────────────────────────────────────────────────
(Intercept)    3.0    0.707107    4.24    0.0132    1.03676    4.96324
x              0.0  NaN         NaN       NaN     NaN        NaN
──────────────────────────────────────────────────────────────────────
julia> lm(@formula(y ~ x), nasty; method=:qr)
LinearModel
y ~ 1 + x
Coefficients:
──────────────────────────────────────────────────────────────────────────────────────────────
                   Coef.   Std. Error                    t  Pr(>|t|)    Lower 95%    Upper 95%
──────────────────────────────────────────────────────────────────────────────────────────────
(Intercept)  7.94411e-16  1.2026e-15                  0.66    0.5561  -3.0328e-15  4.62162e-15
x            1.0e12       0.000362597  2757880273211543.50    <1e-45   1.0e12      1.0e12
──────────────────────────────────────────────────────────────────────────────────────────────
```
but the doc test failed.

Please let us know if we can handle it better way.

Reply:
Yes, I know. I would show them anyway, but just disable doctesting the results.

Reply:
So for this doctring, do we just change three back ticks followed by `jldoctest` to three back ticks followed by `julia`?

Reply:
yes, just not run doctest on this line. (I know it is not the cleanest, but the issue is that we are using linear algebra package that does not produce consistent results on different machines)

Reply:
You can also filter out some values from the doctest, see https://documenter.juliadocs.org/stable/man/doctests/#Filtering-Doctests. That ensure that at least the code runs. Not sure how easy it would be to use here, maybe just something like 
```
```jldoctest; filter = [r"^\(Intercept\).*$", r"^x.*$"]
```
would do?

Reply:
Even after using filters, I am unable to pass the doctest.  One of the reasons is, not only estimates are different, but the number of `-` (or the position of `Coeff.`) is also changed depending on estimates.
 I have changed the example.